File: webflux/simple-crud/src/main/java/br/com/leonardoferreira/poc/Application.java
package br.com.leonardoferreira.poc;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.data.mongodb.config.EnableReactiveMongoAuditing;

@SpringBootApplication
@EnableReactiveMongoAuditing
public class Application {

    public static void main(final String[] args) {
        SpringApplication.run(Application.class, args);
    }

}

File: springfu/src/main/kotlin/br/com/leonardoferreira/Application.kt
package br.com.leonardoferreira

import org.springframework.boot.WebApplicationType
import org.springframework.fu.kofu.application
import org.springframework.fu.kofu.webmvc.webMvc

val app = application(WebApplicationType.SERVLET) {
    beans {
        bean<MyService>()
    }
    webMvc {
        router {
            GET("/") {
                val myService = ref<MyService>()
                ok().body(myService.greeting())
            }
        }
        converters {
            string()
            jackson {
                indentOutput = true
            }
        }
    }
}

fun main() {
    app.run()
}

class MyService {
    fun greeting() = mapOf("ola" to "mundo")
}


data class Time(val id: String, val name: String) {
    init {
        ...
    }
}



